src/bot.ml

```diff
@@ -130,10 +135,62 @@ let extract_commit json =
         sha )
     |> to_string
 
-let gitlab_ref (issue : GitHub_subscriptions.issue) :
-    GitHub_subscriptions.remote_ref_info =
-  { name= f "pr-%d" issue.number
-  ; repo_url= gitlab_repo ~owner:issue.owner ~name:issue.repo }
+let gitlab_ref ~(issue : GitHub_subscriptions.issue) =
+  let gh_repo = issue.owner ^ "/" ^ issue.repo in
+  let open Lwt.Infix in
+  (* First, we check our hashtable for a key named after the GitHub
+     repository and return the associated GitLab repository. If the
+     key is not found, we load the config file from the default branch.
+     Last (backward-compatibility) we assume the GitLab and GitHub
+     projects are named the same. *)
+  ( match gitlab_of_github gh_repo with
+  | None -> (
+      Stdio.printf "No correspondence found for GitHub repository %s/%s.\n"
+        issue.owner issue.repo ;
+      GitHub_queries.get_default_branch ~bot_info ~owner:issue.owner
+        ~repo:issue.repo
+      >>= function
+      | Ok branch -> (
+          GitHub_queries.get_file_content ~bot_info ~owner:issue.owner
+            ~repo:issue.repo ~branch ~file_name:"coqbot.toml"
+          >>= function
+          | Ok (Some content) ->
+              let gl_repo =
+                Option.value
+                  (Config.subkey_value
+                     (Config.toml_of_string content)
+                     "mapping" "gitlab")
+                  ~default:gh_repo
+              in
+              ( match Hashtbl.add gitlab_mapping ~key:gl_repo ~data:gh_repo with
+              | `Duplicate ->
+                  ()
+              | `Ok ->
+                  () ) ;
+              ( match Hashtbl.add github_mapping ~key:gh_repo ~data:gl_repo with
+              | `Duplicate ->
+                  ()
+              | `Ok ->
+                  () ) ;
+              Lwt.return gl_repo
+          | _ ->
+              Lwt.return gh_repo )
+      | _ ->
+          Lwt.return gh_repo )
+  | Some r ->
+      Lwt.return r )
+  >|= fun gl_repo ->
+  let owner, name =
+    match Str.split (Str.regexp "/") gl_repo with
+    | [owner; repo] ->
+        (fun () -> Lwt_io.printf "ok %s %s\n" owner repo) |> Lwt.async ;
+        (owner, repo)
+    | _ ->
+        Stdio.printf "Something unexpected happened.\n" ;
```

Comment:
As a general rule, it's best to make these error messages more explicit (even if they should never be raised). Something like "Something unexpected happened during...".